src/plugins/chart_expressions/expression_heatmap/common/expression_functions/heatmap_function.ts

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import type { DatatableColumn } from '../../../../expressions/public';
+import { ExpressionValueVisDimension } from '../../../../visualizations/common';
+import { prepareLogTable, Dimension } from '../../../../visualizations/common/prepare_log_table';
+import { HeatmapExpressionFunctionDefinition } from '../types';
+import {
+  EXPRESSION_HEATMAP_NAME,
+  EXPRESSION_HEATMAP_GRID_NAME,
+  EXPRESSION_HEATMAP_LEGEND_NAME,
+} from '../constants';
+
+const convertToVisDimension = (columns: DatatableColumn[], accessor: string) => {
+  const column = columns.find((c) => c.id === accessor);
+  if (!column) return;
+  return {
+    accessor: column.id,
+    format: column.meta.params,
+    type: 'vis_dimension',
+  } as unknown as ExpressionValueVisDimension;
+};
+
+const prepareHeatmapLogTable = (
+  columns: DatatableColumn[],
+  accessor: string | ExpressionValueVisDimension,
+  table: Dimension[],
+  label: string
+) => {
+  const dimension =
+    typeof accessor === 'string' ? convertToVisDimension(columns, accessor) : accessor;
+  if (dimension) {
+    table.push([[dimension], label]);
+  }
+};
+
+export const heatmapFunction = (): HeatmapExpressionFunctionDefinition => ({
+  name: EXPRESSION_HEATMAP_NAME,
+  type: 'render',
+  inputTypes: ['datatable'],
+  help: i18n.translate('expressionHeatmap.function.help', {
+    defaultMessage: 'Heatmap visualization',
+  }),
+  args: {
+    // used only in legacy heatmap, consider it as @deprecated
+    percentageMode: {
+      types: ['boolean'],
+      default: false,
+      help: i18n.translate('expressionHeatmap.function.percentageMode.help', {
+        defaultMessage: 'When is on, tooltip and legends appear as percentages.',
+      }),
+    },
+    palette: {
+      types: ['palette'],
+      help: i18n.translate('expressionHeatmap.function.palette.help', {
+        defaultMessage: 'Provides colors for the values, based on the bounds.',
+      }),
+    },
+    legend: {
+      types: [EXPRESSION_HEATMAP_LEGEND_NAME],
+      help: i18n.translate('expressionHeatmap.function.legendConfig.help', {
+        defaultMessage: 'Configure the chart legend.',
+      }),
+    },
+    gridConfig: {
+      types: [EXPRESSION_HEATMAP_GRID_NAME],
+      help: i18n.translate('expressionHeatmap.function.gridConfig.help', {
+        defaultMessage: 'Configure the heatmap layout.',
+      }),
+    },
+    showTooltip: {
```

Comment:
ℹ️ This setting is only supported by Viz editor for now, but could be considered as an addition for Lens. There are users that don't want to have tooltips.